Fix issues with MeasureManager

macumber opened the pull request:
- Add loading tab to discourage selecting another measure which arguments for first measure are being computed (this can take a long time in failure case).  
- Catch all std::exceptions instead of just RubyExceptions which prevents crash.  Fixes #656

macumber commented:
@jmarrec when updating a model with the old results measure we probably need to swap the new one out, currently there is a crash due to this

macumber commented on src/openstudio_app/OpenStudioApp.hpp:
```diff
@@ -275,7 +276,7 @@ class OpenStudioApp : public OSAppBase
   QTranslator m_qtBaseTranslator;
   QString m_currLang;
-  bool m_useClassicCLI;
+  //bool m_useClassicCLI;
```
```suggestion
```

macumber commented:
@jmarrec if you have a chance to review this that would be good, the PR is not too big if you ignore `src/openstudio_app/Resources/` which is just copies of measures from BCL

jmarrec commented:
I'd probably have liked if the "Example > Shoebox Model" was isolated into its own PR.

macumber commented:
> I'd probably have liked if the "Example > Shoebox Model" was isolated into its own PR.

Agreed, sorry I was in a rush, I can move it to a new PR if you'd like

macumber commented on src/openstudio_app/OpenStudioApp.hpp:
```diff
@@ -275,7 +276,6 @@ class OpenStudioApp : public OSAppBase
   QTranslator m_qtTranslator;
   QTranslator m_qtBaseTranslator;
   QString m_currLang;
-  bool m_useClassicCLI;
```
This issue prevents using the classic CLI for the measure manager now: https://github.com/NREL/OpenStudio/issues/5212

I figure it's probably not really worth supporting using classic for the measure manager in the future either, what do you think?  You can still use the classic option for the run though.

macumber commented on src/openstudio_app/Resources/ShoeboxModel/ShoeboxExample/measures/create_bar_from_space_type_ratios/LICENSE.md:
I wanted to make these measures/example models part of the installer, so it is just ready to go with no download needed.    We could fetch the measures/example models from https://github.com/openstudiocoalition/openstudio-coalition-measures at build time and keep them in the build directory rather than checking them in here.  Does that sound better?

macumber replied:
I removed these measures from this repo, they are now pulled from https://github.com/openstudiocoalition/openstudio-coalition-measures at configure time

macumber commented on src/openstudio_app/Resources/ShoeboxModel/ShoeboxExample/measures/openstudio_results/LICENSE.md:
I guess we could add the report measure to the ShoeboxExample model at run time.  I had wanted to keep the model exactly the same as in https://github.com/openstudiocoalition/openstudio-coalition-measures/tree/main/models/ShoeboxExample, that way it would be simple to extend to other models in the future.

macumber replied:
I removed these measures from this repo, they are now pulled from https://github.com/openstudiocoalition/openstudio-coalition-measures at configure time

macumber commented:
Ping @jmarrec, do you want to give this a review?  If not, I think we could merge and push out an RC1.  What do you think?

jmarrec commented:
> Ping @jmarrec, do you want to give this a review? If not, I think we could merge and push out an RC1. What do you think?

An RC1 for 1.8.0? I'd like to see if we can get an updated CLI with https://github.com/NREL/OpenStudio/issues/5230 fixed first

macumber commented:
I don't think we need an updated 1.8.0 package.  After I tested, the main issue was with the OpenStudio Results measure which gets downloaded and added to each model by default at startup.  I fixed that in the 1.7.2 release. The other BCL menus have some filtering built into them, I think I have handled all the needed cases in this PR such that we don't have to have an update to OS SDK.
